src/sentry/services/hybrid_cloud/user/impl.py

```diff
@@ -103,6 +103,18 @@ def update_user(
             User.objects.filter(id=user_id).update(**attrs)
         return self.serialize_many(filter=dict(user_ids=[user_id]))[0]
 
+    def get_user_by_social_auth(
+        self, organization_id: int, provider: str, uid: str
+    ) -> Optional[RpcUser]:
+        user = User.objects.filter(
+            social_auth__provider=provider,
+            social_auth__uid=uid,
+            orgmembermapping_set__organization_id=organization_id,
```

Comment:
This looks good, but we can't merge this until the membermapping backfill is complete.